manifests/tlog.pp

```diff
@@ -1,25 +1,24 @@
 # @summary Setup and configure tlog
 #
+# @param conf
+#   tlog config for /etc/tlog/tlog-rec-session.conf
+#
 # @example
 #   include profile_session_log::tlog
-class profile_session_log::tlog {
-
-  $enabled = lookup("${module_name}::enable_session_log", Boolean)
-  $required_pkgs = lookup("${module_name}::install::required_pkgs", Array)
-
-  if ($enabled) {
+class profile_session_log::tlog (
+  String $conf,
+) {
+  if ($profile_session_log::enable) {
     $ensure_parm = 'present'
   } else {
     $ensure_parm = 'absent'
   }
 
   file { '/etc/tlog/tlog-rec-session.conf':
     ensure  => $ensure_parm,
-    source  => "puppet:///modules/${module_name}/tlog-rec-session.conf",
+    content => $conf,
```

Comment:
Could this be a template with parameters?
How much actually might change on a different system?

Reply:
Hard to say, but it seems like most of the config would stay the same so I made it a template and just set the `message` to be configurable.